docs/source/python/parquet.rst

```diff
@@ -595,3 +595,172 @@ One example is Azure Blob storage, which can be interfaced through the
+
+Configuration of connection to KMS (``kms_connection_config`` used when
+creating file encryption and decryption properties) includes the following
+options:
+
+* ``kms_instance_url``, URL of the KMS instance.
+* ``kms_instance_id``, ID of the KMS instance that will be used for encryption
+  (if multiple KMS instances are available).
+* ``key_access_token``, authorization token that will be passed to KMS.
+* ``custom_kms_conf``, a string dictionary with KMS-type-specific configuration.
+
+Encryption configuration
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Encryption configuration (``encryption_config`` used when creating file
+encryption properties) includes the following options:
+
+* ``footer_key``, ID of the master key for footer encryption/signing.
+* ``column_keys``, list of columns to encrypt, with master key IDs.
+* ``uniform_encryption``, encrypt footer and all columns with the same
+  encryption key.
+* ``encryption_algorithm``, parquet encryption algorithm.
+  Can be ``AES_GCM_V1`` (default), or ``AES_GCM_CTR_V1``.
+* ``plaintext_footer``, write files with plaintext footer.
+* ``double_wrapping``, use double wrapping - where data encryption keys (DEKs)
+  are encrypted with key encryption keys (KEKs), which in turn are encrypted
+  with master keys. If set to false, use single wrapping - where DEKs are
+  encrypted directly with master keys.
+* ``cache_lifetime``, lifetime of cached entities (key encryption keys,
+  local wrapping keys, KMS client objects)
+* ``internal_key_material``, store key material inside Parquet file footers;
+  this mode doesn’t produce additional files. If set to false, key material is
+  stored in separate files in the same folder, which enables key rotation for
+  immutable Parquet files.
+* ``data_key_length_bits``, length of data encryption keys (DEKs), randomly
+  generated by parquet key management tools. Can be 128, 192 or 256 bits.
+
+.. note::
+   By default, Parquet implements a "double envelope encryption" mode, that
+   minimizes the interaction of the program with a KMS server. In this mode,
+   the DEKs are encrypted with "key encryption keys" (KEKs, randomly generated
+   by Parquet). The KEKs are encrypted with MEKs in KMS; the result and the
+   KEK itself are cached in the process memory. Users interested in regular
+   envelope encryption, can switch to it by setting the double_wrapping
+   parameter of EncryptionConfiguration to false.
+
+An example encryption configuration:
+
+.. code-block:: python
+
+   encryption_config = pq.EncryptionConfiguration(
+      footer_key="footer_key_name",
+      column_keys={
+         "column_key_name": ["Column1", "Column2"],
+      },
+   )
+
+Decryption configuration
+~~~~~~~~~~~~~~~~~~~~~~~~
+   
+Decryption configuration (``decryption_config`` used when creating file
+decryption properties) is optional and it includes the following options:
+
+* ``cache_lifetime``, lifetime of cached entities (key encryption keys, local
+  wrapping keys, KMS client objects).
```

Comment:
In which unit is the lifetime expressed?

Reply:
It's a datetime.timedelta. Adding it to the description.